homeassistant/components/tts/entity.py

```diff
@@ -198,6 +198,8 @@ async def async_get_tts_audio(
 
         Return a tuple of file extension and data as bytes.
         """
+        self.__last_tts_loaded = dt_util.utcnow().isoformat()
+        self.async_write_ha_state()
```

Comment:
This isn't correct. I think we need to bring back the internal method that was removed in #148946. Platforms are meant to override `async_get_tts_audio`. If they do that, the datetime state won't be set correctly unless we have a separate method that does this. As far as I can tell, the state isn't updated anywhere else for this case.

Reply:
We're probably missing proper tests too for this, since they didn't fail.

Reply:
Thanks! I've restored the internal method which actually simplified the other code :+1:

Reply:
Please also add a test for an entity that implements `async_get_tts_audio` and doesn't support streaming. Check that the state is updated correctly when audio is returned.

The test should fail if if we eg move the state write method from the internal method to `async_get_tts_audio`.

Reply:
I added the new test, but made it a bit more direct. I just call both the internal and external `get_tts_audio` methods (non-streaming) of the base class and ensure only the internal one changes the state.

Reply:
Where do we test that an entity that doesn't support streaming calls `async_internal_get_tts_audio`?

Reply:
Added an explicit test for this.